core/domain/exp_domain.py

```diff
@@ -643,6 +650,11 @@ def validate(self, strict=False):
             raise utils.ValidationError(
                 'Invalid language_code: %s' % self.language_code)
 
+        if not isinstance(self.image_id_counter, int):
+            raise utils.ValidationError(
+                'Expected image_id_counter to be a integer, received %s' %
+                self.image_id_counter)
```

Comment:
we should pass this value to the state.validate and check all the image_id are less than image_counter.

Reply:
done